brave-core/src/main/java/com/github/kristofa/brave/LocalTracer.java

```diff
@@ -83,7 +83,10 @@ public SpanId startNewSpan(String component, String operation) {
     }
 
     private SpanId getNewSpanId() {
-        Span parentSpan = spanAndEndpoint().state().getCurrentServerSpan().getSpan();
+        Span parentSpan = spanAndEndpoint().state().getCurrentLocalSpan();
```

Comment:
one of the joys of having hidden constructors is we can change them :)

For example, LocalTracer's Builder is hidden. You can add a `Builder allowNestedSpans(boolean allowNestedSpans)` internally to brave, and the only thing that would create it is Brave's instantiation logic who can make an instanceof check. I'd put a TODO referencing #166